packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -388,7 +388,26 @@ abstract class Route<T> {
   @mustCallSuper
   void changedExternalState() { }
 
-  /// Discards any resources used by the object.
+  bool _holdingGracefulDisposal = false;
+  /// Set whether [gracefullyDispose] should call [dispose] at a later time.
+  ///
+  /// Calling [markHoldingGracefulDisposal] with false will invoke [dispose] if
+  /// [gracefullyDispose] has been called while [markHoldingGracefulDisposal] is
+  /// set to true. See [gracefullyDispose] for detailed usage.
+  @protected
+  void markHoldingGracefulDisposal(bool value) {
```

Comment:
Can this be a modal route only thing, where the Navigator still calls disposed on route but the modal route will override the dispose to do a gracefully dispose?